### PR TITLE
FIX Fix CI builds, remove excessive Travis builds matrixes, update composer constraint for gridfield-bulk-tools

### PR DESCRIPTION
**Pull request by robbieaverill**
Slight changes to the Travis configuration: run coverage upload only if it succeeds, remove Scrutinizer from it (it's not expecting to receive it anyway), cleaned up the build matrix

### CODE REVIEW

**Comment by robbieaverill** [comment]
@dhensby this is good to go now. If you're happy with it it might be worth squashing it

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 language: php
 
-sudo: true
```
Comment (dhensby) [inline .travis.yml]: don't we need `sudo: false`? (OK - looked at docs, you only need it if the repo was enabled before 20:15 - could still be nice to be explicit... maybe?)

File: composer.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     }],
     "require": {
         "silverstripe/framework": "^4.0@dev",
-        "colymba/gridfield-bulk-editing-tools": "~2.1"
+        "colymba/gridfield-bulk-editing-tools": "^3.0@dev"
```
Comment (dhensby) [inline composer.json]: Why do we want to explicitly include a dev version?

File: composer.json [hunk 3]
```diff
@@ -29,5 +29,7 @@
             "SilverStripe\\Comments\\Tests\\": "tests/"
         }
     },
-    "license": "BSD-3-Clause"
+    "license": "BSD-3-Clause",
+    "minimum-stability": "dev",
+    "prefer-stable": true
```
Comment (dhensby) [inline composer.json]: these couple of lines only work in the root `composer.json` (https://getcomposer.org/doc/04-schema.md#minimum-stability)

File: .travis.yml (hunk 1 above)
Comment (robbieaverill) [inline .travis.yml]: Happy to put it back but figured since it's not relevant it should be removed

File: composer.json (hunk 2 above)
Comment (robbieaverill) [inline composer.json]: Without this it pulls 2-x- goes along with the minimum stability flag - they were both needed when I was testing it on my fork. They're only temporary until we have an alpha tag released anyway

Comment (dhensby) [inline composer.json]: the requirement of ^3.0 pulls in 2.x? that's got to be a bug... What version of composer are you using?

Comment (robbieaverill) [inline composer.json]: I've had a beer. I could be wrong. I'll push a comnit without the specific and see what happens.

Comment (robbieaverill) [inline composer.json]: Seems good - very nice!!

**Pull request merged**